projects.markdown

```diff
@@ -12,13 +12,47 @@ irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat
 nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa
 qui officia deserunt mollit anim id est laborum. 
 
-## Typed Compilation of Dependent Types 
+## Typed Compilation of Dependent Types
+Dependently typed languages such as Coq have been used to specify and prove the full
+functional correctness of software such as the CompCert optimizing C compiler ([Leroy 2009][#leroy2009]), the
+CertiKOS operating system kernel ([Gu et al. 2015][#gu2015], [Gu et al. 2016][#gu2016]), and the
+implementation of cryptographic primitives and protocols ([Barthe et al. 2009][#barthe2009], [Appel et
+al. 2015][#appel2015]).
 
-unde omnis iste natus error sit voluptatem accusantium doloremque laudantium,
-totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi
-architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia
-voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores
-eos qui ratione voluptatem sequi nesciunt.
+Unfortunately, these verification efforts are all easily undone when linking with external components.
+Typically, Coq programs are translated to OCaml, a high-level functional language with a strong type
+system (but without dependent types).
+This translation discards the Coq specifications, so OCaml can no longer enforce the
+invariants Coq relied on when verifying the source program.
+It is simple to construct a small verified Coq function that, when compiled and linked with well-typed
+code in OCaml, promptly *jumps to an arbitrary location in memory*.
+
+The key to preventing programs from being linked with ill-behaved external components is
+*type-preserving compilation*.
+Types express interfaces, invariants, and specifications.
+A type-preserving compiler preserves all typing information from the source program into the target
+programs.
+Safe linking is guaranteed through type checking in the target language.
+
+Type preservation for dependent types is a fundamentally hard problem and has been stalled for fifteen
+years.
+Dependent type theory has been used a foundation for mathematics analogous to set theory.
+Every standard compiler design decision must also avoid running afoul of fundamental problems such as
+Girard's Paradox (the type-theoretic version of Russel's Paradox) and Gödel's incompleteness
+theorem.
+An impossibility result fifteen years ago showed that a known verified type-preserving compiler
+pass---the standard typed CPS translation---leads to Girard's paradox when used on dependently typed
+languages ([Barthe and Uustalu 2002][#barthe2002]).
+
+Our group has recently developed alternative type-preserving versions of the two major front-end
+compiler passes, including an alternative CPS translation that avoids Girard's paradox.
```

Comment:
Should we link to this paper? Probably we should decide on a policy for that for the whole page.

Reply:
Yes, I think we should always link to papers.